crates/interface-types/src/lib.rs

```diff
@@ -66,17 +67,41 @@ impl ModuleData {
         // find the right section.
         let mut reader = wasmparser::ModuleReader::new(wasm)?;
         let mut found = false;
+        let mut wasi_module_name = None;
         while !reader.eof() {
             let section = reader.read()?;
-            if let wasmparser::SectionCode::Custom { name, .. } = section.code {
-                if name == "webidl-bindings" {
-                    found = true;
-                    break;
+
+            match section.code {
+                wasmparser::SectionCode::Custom { name, .. } => {
+                    if name == "webidl-bindings" {
+                        found = true;
+                        break;
+                    }
+                }
+
+                // If we see the import section then see if we can find a wasi
+                // module import which we can later use to register the wasi
+                // implementation automatically.
+                wasmparser::SectionCode::Import => {
+                    let section = section.get_import_section_reader()?;
+                    for import in section {
+                        let import = import?;
+                        match import.module {
+                            "wasi" | "wasi_unstable" => {
```

Comment:
Note that this logic was lifted up from below to work for modules that don't have an interface types section as well.

Reply:
`wasi_snapshot_*` will soon be becoming a thing, so we should add `.starts_with("wasi_snapshot_*")` here too.

Reply:
Sure yeah, I went ahead and refactored this a bit as well to hopefully future proof slightly. I think to support multiple wasi interfaces we'll need a bit more logic with wasmtime-wasi anyway.